hibernate-core/src/main/java/org/hibernate/loader/ast/internal/LoaderSelectBuilder.java

```diff
@@ -133,6 +133,45 @@ public static SelectStatement createSelect(
 		return process.generateSelect();
 	}
 
+	/**
+	 * Create an SQL AST select-statement for a select by unique key based on matching one-or-more keys
+	 *
+	 * @param loadable The root Loadable
+	 * @param partsToSelect Parts of the Loadable to select.  Null/empty indicates to select the Loadable itself
+	 * @param restrictedPart Part to base the where-clause restriction on
+	 * @param cachedDomainResult DomainResult to be used.  Null indicates to generate the DomainResult
+	 * @param numberOfKeysToLoad How many keys should be accounted for in the where-clause restriction?
+	 * @param loadQueryInfluencers Any influencers (entity graph, fetch profile) to account for
+	 * @param lockOptions Pessimistic lock options to apply
+	 * @param jdbcParameterConsumer Consumer for all JdbcParameter references created
+	 * @param sessionFactory The SessionFactory
+	 */
+	public static SelectStatement createSelectByUniqueKey(
+			Loadable loadable,
+			List<? extends ModelPart> partsToSelect,
+			ModelPart restrictedPart,
+			DomainResult<?> cachedDomainResult,
+			int numberOfKeysToLoad,
+			LoadQueryInfluencers loadQueryInfluencers,
+			LockOptions lockOptions,
+			Consumer<JdbcParameter> jdbcParameterConsumer,
+			SessionFactoryImplementor sessionFactory) {
+		final LoaderSelectBuilder process = new LoaderSelectBuilder(
+				sessionFactory,
+				loadable,
+				partsToSelect,
+				Arrays.asList( restrictedPart ),
```

Comment:
Using `Collections.singleton()` would be more natural and readable. JMH benchmark should show its performance advantage as well (though not tried yet).

Reply:
The article (https://dzone.com/articles/singleton-list-showdown-collectionssingletonlist-v) shows the `Arrays.asList()` is twice as slow than `Collections.singletonList()`.

Reply:
Thanks @NathanQingyangXu , i'm going to use your suggestion